lib/ae_mdw/db/name.ex

```diff
@@ -233,16 +234,31 @@ defmodule AeMdw.Db.Name do
   #
   # Private functions
   #
-  defp claimed_by(m_name) do
-    [{{_, _}, last_claim_txi} | _] = Model.name(m_name, :claims)
-    %{tx: %{account_id: orig_owner}} = Format.to_raw_map(read_tx!(last_claim_txi))
-    {:id, :account, first_owner} = orig_owner
-    first_owner
+  defp pointee_at(m_name, ref_txi) do
+    m_name
+    |> Model.name(:updates)
+    |> find_update_txi_before(ref_txi)
+    |> case do
+      nil ->
+        Log.warn("name #{elem(m_name, 1)} without pointer at txi #{ref_txi}!")
```

Comment:
We can start pattern-matching on records instead of using `elem` or the getter functions like `Model.name(:updates)`.` This can be done like so:

```elixir
def pointee_at(Model.name(updates: updates, index: index), ref_txi)
```

This way you would get compiler errors if the field is wrongly spelled or doesn't exist, plus some added clarity.

Reply:
I am pro pattern matching in many cases. This is a good one even though `Model.name/2` also detects typos.